tasks/analysis.py

```diff
@@ -417,7 +418,23 @@ def process_impact_result(self, impact_result, analysis_id):
             impact_result['output'].get('analysis_summary'))
 
         # generate report when analysis has ran successfully
-        async = generate_report.delay(impact_url)
+        custom_template_path = None
+        setting_template_path = settings.LOCALIZED_QGIS_REPORT_TEMPLATE.get(
+            analysis.language_code)
+        if os.path.exists(setting_template_path):
+            filename = os.path.basename(setting_template_path)
+            custom_template_path = os.path.join(
+                settings.INASAFE_LAYER_DIRECTORY_BASE_PATH, filename)
+            if not os.path.exists(custom_template_path):
+                shutil.copy(setting_template_path, custom_template_path)
```

Comment:
here I was using mounted directory between django and headless as defined in `settings.INASAFE_LAYER_DIRECTORY_BASE_PATH` where it stores impact layer from `file_upload`. do you mean I need to copy the template to the actual analysis output directory of headless?

Reply:
I propose we avoid race conditions by using `open` instead of `exists`. because if we want to use the analysis output directory on headless we need to run the analysis first, yet we need to define the custom template before we run the analysis.

Reply:
@myarjunar 
Ok, let me explain first what I meant.
There is no folder structure at ```settings.INASAFE_LAYER_DIRECTORY_BASE_PATH``` 
Let's say ```settings_template_path = '/path/to/custom.qpt'```
That means, ```filename = 'custom.qpt'```
This would make: ```custom_template_path = '/usr/src/app/geonode/qgis_layer/custom.qpt'```
for every analysis made. Race condition will occur if there are two headless analysis running with different locale, but is using this same file. 

So, my first suggestion there is to use actual analysis output directory, because each analysis will have different folder. So, if you run two analysis: the templates will be a different file and safe from race condition, because it is on a different folder.

```
/home/geosafe/impact_layers/ANALYSIS_1/custom.qpt
/home/geosafe/impact_layers/ANALYSIS_2/custom.qpt
```

Reply:
You can directly uses the folder from 
```os.path.dirname(impact_result['output'].get('analysis_summary'))```
because that's the analysis output directory.